### PR TITLE
Fixes #33: server documentation not well generated

### LINKED ISSUE #33: Server command parameter documentation not well generated
PyTango servers don´t get their documentation well generated.
Generated sphinx comes with lots of artifacts

### PR DESCRIPTION
**Pull request by tiagocoutinho**

### CODE REVIEW

File: src/boost/python/server.py [hunk 1]
```diff
@@ -1088,11 +1126,12 @@ def Pressurize(self, pressure):
             green_mode=green_mode)
     name = f.__name__
 
-    dtype_in, dformat_in = _get_tango_type_format(dtype_in, dformat_in)
-    dtype_out, dformat_out = _get_tango_type_format(dtype_out, dformat_out)
 
-    din = [from_typeformat_to_type(dtype_in, dformat_in), doc_in]
-    dout = [from_typeformat_to_type(dtype_out, dformat_out), doc_out]
+    t_dtype_in, t_dformat_in = _get_tango_type_format(dtype_in, dformat_in)
+    t_dtype_out, t_dformat_out = _get_tango_type_format(dtype_out, dformat_out)
```
Comment (vxgmichel) [inline src/boost/python/server.py]: Can we have more meaningful names for those variables? Or keep them packed as `typeformat_in` and `typeformat_out`?

File: src/boost/python/server.py [hunk 2]
```diff
@@ -1102,6 +1141,15 @@ def cmd(self, *args, **kwargs):
...
 
     cmd.__tango_command__ = name, [din, dout]
+
+    # try to create a minimalistic __doc__
+    if cmd.__doc__ is None:
+        try:
+            cmd.__doc__ = __build_command_doc(f, name, dtype_in, doc_in,
+                                              dtype_out, doc_out)
+        except Exception as e:
+            pass
```
Comment (vxgmichel) [inline src/boost/python/server.py]: What about `cmd.__doc__ = 'TANGO command'` if the documentation can't be built?

File: src/boost/python/server.py [hunk 3]
```diff
@@ -838,6 +840,8 @@ def get_attribute(self, obj):
     # --------------------
 
     def __get__(self, obj, objtype):
+        if obj is None:
+            return self
```
Comment (vxgmichel) [inline src/boost/python/server.py]: :+1:

**Comment by vxgmichel** [comment]
Looks good! Can we merge it in `prepare-tango9-release` though?

File: src/boost/python/server.py (hunk 2 above)
Comment (tiagocoutinho) [inline src/boost/python/server.py]: Agree

File: src/boost/python/server.py (hunk 1 above)
Comment (tiagocoutinho) [inline src/boost/python/server.py]: ok, I will keep them packed

**Comment by tiagocoutinho** [comment]
Ok, changes have been done.

BTW, Since I did an amend to my commit, I see that your comments have been marked as *outdated*.
For reference do you think I should have done a second commit instead?

**Comment by vxgmichel** [comment]
@tiagocoutinho Github tags a diff as outdated if it doesn't match the current state of the branch, so amending or making new commits shouldn't make a difference, though it might be easier to follow the timeline if new commits are made. Also I don't mind having a lot of commits in a PR since I usually squash them before merging (github lets you do that directly from the web interface).

**Comment by vxgmichel** [comment]
Tested, it seems to work fine! Can I merge it @tiagocoutinho, or do you plan to add more to this PR?

**Comment by tiagocoutinho** [comment]
After testing it better now it seems to finally work.
You have my Ok to merge. Thanks for all the comments

**Pull request merged**